src/test/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorIntegrationTest.java

```diff
@@ -0,0 +1,392 @@
+                (a, e) -> {
+                    // We need to filter out sites where we saw a DP of 250 because we are comparing the results to GATK3, which downsamples to 250 reads per sample, which GATK4 does not currently support.
+                    if (!e.getGenotypes().stream().anyMatch(g -> g.hasDP() && g.getDP() >= 250)) {
+                        VariantContextTestUtils.assertVariantContextsAreEqualAlleleOrderIndependent(a, e, ATTRIBUTES_TO_IGNORE, header);
+                    }
+                },
+                b37_reference_20_21);
+    }
+
+    @Test
+    public void testNoAnnotsNotAsking1() throws IOException {
+        final File expected = new File(getToolTestDataDir() + "expected/testHsAnnotsNotAsking1.vcf");
+        final VCFHeader header = getHeaderFromFile(expected);
+        runVariantAnnotatorAndAssertSomething(getTestFile("vcfexamplemultisampleempty.vcf"), new File(getToolTestDataDir() + "expected/testHasNoAnnotsNotAsking1.vcf"), Arrays.asList( "-I", largeFileTestDir + "CEUTrio.multisample.b37.1M-1M50k.bam"),
+                (a, e) -> {
+                    // We need to filter out sites where we saw a DP of 250 because we are comparing the results to GATK3, which downsamples to 250 reads per sample, which GATK4 does not currently support.
+                    if (!e.getGenotypes().stream().anyMatch(g -> g.hasDP() && g.getDP() >= 250)) {
+                        VariantContextTestUtils.assertVariantContextsAreEqualAlleleOrderIndependent(a, e, ATTRIBUTES_TO_IGNORE, header);
+                    }
+                },
+                b37_reference_20_21);
+    }
+
+
+    @Test
+    public void testNoAnnotsAsking1() throws IOException {
+        final File expected = new File(getToolTestDataDir() + "expected/testHasNoAnnotsAsking1.vcf");
+        final VCFHeader header = getHeaderFromFile(expected);
+        runVariantAnnotatorAndAssertSomething(getTestFile("vcfexamplemultisampleempty.vcf"), new File(getToolTestDataDir() + "expected/testHasNoAnnotsAsking1.vcf"), Arrays.asList("-G", "Standard", "-I", largeFileTestDir + "CEUTrio.multisample.b37.1M-1M50k.bam"),
+                (a, e) -> {
+                    // We need to filter out sites where we saw a DP of 250 because we are comparing the results to GATK3, which downsamples to 250 reads per sample, which GATK4 does not currently support.
+                    if (!e.getGenotypes().stream().anyMatch(g -> g.hasDP() && g.getDP() >= 250)) {
+                        VariantContextTestUtils.assertVariantContextsAreEqualAlleleOrderIndependent(a, e, ATTRIBUTES_TO_IGNORE, header);
+                    }
+                },
+                b37_reference_20_21);
+    }
+    @Test
+    public void testOverwritingHeader() throws IOException {
```

Comment:
it is using updatedContigs actually, everything is I believe. The reason I used it is because it was the best way to match the header to gatk3 since that is what it was using.

Reply:
What is our sequence dictionary validation policy for VCFs?  As things are now, there are a lot of opportunities for mayhem.  For example, CalculateGenotypePosteriors doesn't need a reference and there's no way to check that the input VCF and the supporting VCF (if specified) are on the same reference.  Ditto VariantRecalibrator.  I believe those tools both check alleles for overlapping variants, but that doesn't mean is 100% secure against input mismatches.